trestle/core/jinja.py

```diff
@@ -177,3 +178,53 @@ def parse(self, parser):
         top_level_output = local_parser.parse()
 
         return top_level_output.body
+
+
+class MDDatestamp(TrestleJinjaExtension):
+    """Inject the parameter of the tag as the resulting content."""
+
+    tags = {'md_datestamp'}
+
+    def __init__(self, environment: Environment) -> None:
+        """Ensure enviroment is set and carried into class vars."""
+        super().__init__(environment)
+
+    def parse(self, parser):
+        """Execute parsing of md token and return nodes."""
+        kwargs = None
+        count = 0
+        while parser.stream.current.type != lexer.TOKEN_BLOCK_END:
+            count = count + 1
+            token = parser.stream.current
+            if count > self.max_tag_parse:
+                raise err.TrestleError(f'Unexpected Jinja tag structure provided at token {token.value}')
+            if token.test('name:md_datestamp'):
+                parser.stream.expect(lexer.TOKEN_NAME)
+            elif kwargs is not None:
+                arg = token.value
+                next(parser.stream)
+                parser.stream.expect(lexer.TOKEN_ASSIGN)
+                token = parser.stream.current
+                exp = self.parse_expression(parser)
+                kwargs[arg] = exp.value
+            else:
+                if parser.stream.look().type == lexer.TOKEN_ASSIGN or parser.stream.look().type == lexer.TOKEN_STRING:
+                    kwargs = {}
+                continue
+
+        if kwargs is not None:
+            if 'format' in kwargs and type(kwargs['format'] is str):
+                date_string = date.today().strftime(kwargs['format'])
+            else:
+                date_string = date.today().strftime('%m/%d/%Y')
+            if 'newline' in kwargs and kwargs['newline'] is False:
```

Comment:
The repeated references to the timestring might be better as a constant - const.JINGA_DATESTAMP_FORMAT or something.  There are many refs to it here and the test.

Reply:
done